certification/engine/engine.go

```diff
@@ -32,8 +32,13 @@ type PolicyRunner interface {
 	ExecutePolicies()
 	// StorePolicies(...[]certification.Policy)
 	Results() runtime.Results
+	Logs() LogMap
 }
 
+// LogMap is a map of logfiles to write (the key), and the data to write to
+// it (the byte slice).
+type LogMap = map[string][]byte
```

Comment:
The code looks good to me. I have a question regarding the design: the reason we do not use the standard Go logging framework is because we want to output the logs to multiple files? Is there any other reason too?

Reply:
I think we could use either the standard log framework, or logrus as an implementation!

The main reason I don't directly use either in the implementations of each policy's `Validate()` (and instead return log information to be written further upstream) is to avoid having each policy handle log output and writing independently. The current design tries to have a single time to "write logs to disk". At some point in the future, I also want to buffer the writes to avoid crashing if we can't write, for example.

An alternate approach would have been to pass in contextual loggers into the Engine, and a further-contextual logger into each Policy validation. That would work as well.

Reply:
I think passing the log object (excuse my Java terminology :D still trying to figure out the GO terminology) as an argument will simplify the design and we don't need to keep track of this map. If the multi threading execution of the policies are a concern, we can append all logs to a byte slice, inside the Validate() method and write it to Log at the end of the function, just before returning.

Reply:
Okay, I'll close this and reopen shortly.